src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -56,7 +56,7 @@ export class TerminalService extends AbstractTerminalService implements ITermina
 
 		this._terminalTabs = [];
 		this._configHelper = this._instantiationService.createInstance(TerminalConfigHelper);
-
+		this.toDispose.push(this.onTerminalNotSplit(() => this._notificationService.warn(nls.localize('terminal.minWidth', "Not enough space to split terminal."))));
```

Comment:
You could avoid doing the event/disposable changes by making `common/terminalService` have some `protected abstract` method similar to `_showTerminalCloseConfirmation` and implement it in `electron-browser/terminalService`